### ISSUE TITLE
ctm, ctm_2site and ctm_split discard their convergence flag, so ipeps() cannot tell a converged environment from an exhausted max_iter

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#839]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

The dense CTM entry points in `ipeps_ctm_convergence.py` compute a `converged` flag and an iteration count, then throw both away. Callers — including the public `ipeps()` — cannot distinguish an environment that converged from one that silently exhausted `max_iter`.

## The code

`ctm_2site` (line 200) carries both through its `while_loop` and discards them into `_`:

```python
# Carry: (env_A, env_B, prev_sv_A, prev_sv_B, iteration, converged)
...
converged = jnp.maximum(diff_A, diff_B) < conv_tol
return (eA, eB, sv_A, sv_B, iteration + 1, converged)

env_A, env_B, _, _, _, _ = jax.lax.while_loop(cond_fn, body_fn, init_carry)
return env_A, env_B
```

`ctm` (line 69) does the same:

```python
env, _, _, _ = jax.lax.while_loop(cond_fn, body_fn, init_carry)
return env
```

`ctm_split` (line 367) reaches the same end by a different route — a Python loop that `break`s on the tolerance and records nothing:

```python
if float(diff) < config.conv_tol:
    break
...
return env
```

**`ctm` and `ctm_2site` are both public API** (`tenax.ctm`, `tenax.ctm_2site`, both in `__all__`). So is `ipeps()`, which calls `ctm_2site` at `ipeps.py:452` and returns `float(energy), (A, B), (env_A, env_B)` — an energy from an environment of unknown convergence status, with no channel to report it.

## The contrast is in-repo

The Tensor-path CTM already does this correctly: `_ctm_tensor_convergence.ctm_tensor` returns `(env, max_eps)`, and callers write `env, _ = ctm_tensor(...)`. The dense path is the outlier, not the convention.

## Evidence that it bites

Measured on `tests/_su_fixtures.physical_su_d2` (see the retraction on #838 for full context — the *state* there was unconverged, which is exactly why this matters):

| conv_tol | E @ chi=6 |
|---|---|
| 1e-10 | −0.541373269118908 |
| 1e-11 | −0.541373269118908 |
| 1e-12 | −0.541373269118908 |

Bit-identical. Meanwhile varying `max_iter` at fixed `conv_tol` moved E by ~7e-3 (100 → −0.541373, 120 → −0.548673, 300 → −0.543002).

Tightening the tolerance doing nothing while the iteration budget changes the answer means that loop was terminating on `max_iter`, not on its criterion — and nothing in the return value said so. That is the whole issue in one table: the caller received a number from a sweep that never met its own convergence test, and had no way to find out.

This is the defect class of #801 and #824 — a gradient path that reported no residual, so an unconverged solve was indistinguishable from a converged one. Same shape, forward side.

## Suggested fix

Return the diagnostics rather than discarding them, following the Tensor path's existing convention.

Both functions are exported, so changing the return arity is breaking. Two options, in preference order:

1. **Opt-in kwarg** — `return_meta: bool = False`, returning `(env, meta)` when set. Non-breaking, and lets `ipeps()` and the benchmark drivers ask for it without touching user code.
2. **Always return a NamedTuple** with `.env` / `.converged` / `.n_iter`, which reads better but breaks every existing call site.

Whichever, `ipeps()` should surface it — a public entry point returning an energy silently derived from a non-converged environment is the actual user-facing harm here.

Worth deciding at the same time: whether a non-converged CTM should **warn** by default. `check_ctm_env` (#747) already established the precedent of warning on an unhealthy environment rather than failing silently, and this is the same argument for a different failure mode.

## Note on the convergence criterion itself

Out of scope for this issue, but adjacent and worth recording: the criterion watches corner singular values (`_ctm_sv_diff(sv, prev_sv)`), never the energy. #780/#781 established that CTM convergence criteria on gauge-dependent quantities can read 1.004 where a gauge-invariant measure reads 1.1e-16. Whether the corner-SV criterion is the right one is a separate question from whether its verdict should be reported — this issue is only about the latter.
